src/System.Data.Common/src/System/Data/Common/DbDataReaderExtensions.cs

```diff
@@ -4,7 +4,7 @@
 
 namespace System.Data.Common
 {
-    public static class DbDataReaderExtensions
+    public static partial class DbDataReaderExtensions
```

Comment:
It also seems like a good idea to add it to the DbDataReaderExtensions class that's in DbDataReaderExtensions.Facade.cs, even if that file doesn't seem to be included for compilation in the csproj.

Reply:
done

Reply:
Sorry if I missed something, but what is the motivation to not add the extension methods directly here?

Reply:
I unresolved this thread because I still think we should consider adding the extension methods directly on the new sponsor class unconditionally, not just under netcoreapp.cs

Reply:
@divega, could you please clarify what you meant by unconditionally?

I added these extensions in `DbDataReaderExtensions.netcoreapp.cs` to easily include/exclude them based on the build target; these APIs are for .NET Core only as far as I understood.

Reply:
@divega if I understand you correctly, now that these methods are extension methods (and not in DbDataReader), they can be added everywhere, since there's no type forwarding to an in-box version, correct?

If so, would we need any additional review/validation to do that?

Reply:
@roji correct. I will take care of asking, but I cannot think of any problem with this. 

@TheBlueSky please add them on the new DbDataReaderExtensions.cs.

Reply:
I added `DataReaderExtensions` under `System.Data` namespace.